### ISSUE TITLE
Add regex rule

### ISSUE THREAD

**Issue by CodyMcMichael** [lifechurch/validstate#12]
regex - Checks to make sure the value matches a given regex

example validation config:

```
shoppingCart: {
  couponCode: { regex: /^(?=\S+[a-zA-Z])(?=\S+\d)\S{8,8}$/ }
}
```

- _Returns boolean_

**Comment by geekiam23** [comment]
Hey Cody, I'd love to take a shot at this. Is it still open?

**Comment by CodyMcMichael** [comment]
@geekiam23 Yeah we'd love some help on it. Go for it! Here are our [contributing guidlines](https://github.com/lifechurch/validstate/blob/master/CONTRIBUTING.md#sending-a-pull-request). Thanks for your help!

**Comment by geekiam23** [comment]
@CodyMcMichael Hey, are you on the Slack channel? I have a couple of questions for you.
